user/ocaml/template.py

```diff
@@ -0,0 +1,60 @@
+pkgname = "ocaml"
+pkgver = "5.3.0"
+pkgrel = 0
+build_style = "gnu_configure"
+configure_script = "configure"
+configure_args = [
+    "--libdir=/usr/lib/ocaml/",
+    "--disable-installing-source-artifacts",
+    "--enable-ocamltest"
+]
+configure_gen = []
+make_dir = "."
+make_check_target = "tests"
+makedepends = ["zstd-devel"]
+depends = ["zstd-devel"]
+pkgdesc = "Main implementation of the OCaml programming language"
+license = "LGPL-2.1-only"
+url = "https://ocaml.org"
+source = f"https://github.com/ocaml/ocaml/archive/refs/tags/{pkgver}.tar.gz"
+sha256 = "eb9eab2f21758d3cfb1e78c7f83f0b4dd6302824316aba4abee047a5a4f85029"
+tools = {"ASPP": "cc -c", "AS": "cc -c"}
+# disable usban integer checks due to camlInternalFormatBasics module failing to build
+hardening = ["!int"]
+
+# Have ocaml compile native binaries on architectures it supports
```

Comment:
is this needed at all

won't it autodetect the right thing

Reply:
also i'm pretty sure we can have native compilation for ppc64, and i'm pretty sure ocaml will require a bunch of patching to handle some of the architectures, so i'll need to check that myself on the target machines